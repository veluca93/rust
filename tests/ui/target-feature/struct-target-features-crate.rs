//@ only-x86_64
//@ aux-build: struct-target-features-crate-dep.rs
//@ check-pass
#![feature(target_feature_11)]
#![feature(struct_target_features)]
//~^ WARNING the feature `struct_target_features` is incomplete and may not be safe to use and/or cause compiler crashes

extern crate struct_target_features_crate_dep;

#[target_feature(enable = "avx")]
fn avx() {}

#[target_feature(from_args)]
fn f(_: struct_target_features_crate_dep::Avx) {
    avx();
}

fn g(_: struct_target_features_crate_dep::NoFeatures) {}

fn main() {
    if is_x86_feature_detected!("avx") {
        let avx = unsafe { struct_target_features_crate_dep::Avx {} };
        f(avx);
    }
    g(struct_target_features_crate_dep::NoFeatures {});
}
