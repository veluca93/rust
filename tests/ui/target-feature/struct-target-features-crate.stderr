warning: the feature `struct_target_features` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/struct-target-features-crate.rs:5:12
   |
LL | #![feature(struct_target_features)]
   |            ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #129107 <https://github.com/rust-lang/rust/issues/129107> for more information
   = note: `#[warn(incomplete_features)]` on by default

warning: 1 warning emitted

